### PR TITLE
Run all performance tests with Java 17 by default

### PR DESCRIPTION
**Pull request by eskatos**
Java 17 is the latest LTS and it is known to be faster and better at memory management.

While Gradle supports being run with Java 8 we should recommend using the latest LTS.
Running our performance tests against older Java versions doesn't make much sense and would make us potentially optimise for the wrong target.

----

The default changed from Java 8 to Java 11 in
* https://github.com/gradle/gradle/pull/23718

### CODE REVIEW

**Comment by eskatos** [comment]
@bot-gradle test APT

**Review by wolfs — APPROVED** [review]
Looks good!

File: subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/generator/FileContentGenerator.groovy [hunk 1]
```diff
@@ -68,7 +68,7 @@ abstract class FileContentGenerator {
 
         dependencies{
         ${
-            language == Language.GROOVY ? directDependencyDeclaration('implementation', 'org.codehaus.groovy:groovy:2.5.8') : ""
```
[the remark this answers is not in the document]
Comment (eskatos) [reply subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/generator/FileContentGenerator.groovy:71]: Good catch, I mixed my mind. Fixed

**Comment by eskatos** [comment]
@bot-gradle test APT

**Pull request merged**